Dockerfile

```diff
@@ -21,9 +21,10 @@ WORKDIR /workspace
 
 USER root
 # Install EPEL repository directly and then ZeroMQ, as epel-release is not in default repos.
+# Install all necessary dependencies including Python 3.9 for chat-completions templating.
```

Comment:
I really think you should revisit the way you're handling dependencies.
This is not maintainable over time and not reproducible in a local environment (I mean for the deps that are installed in github workflows).

I really think you should switch to how we do it in GIE, where depedencies are downloaded to a local bin directory, and both CI, makefile and local testing are all referencing to that dir.

Reply:
yes. it was a non blocking comment. just something to keep in mind.
might be worth opening an issue to track it